tests/foreach/foreach.t

```diff
@@ -0,0 +1,95 @@
+SETUP PYTHON
+
+$ [ -n "$PYTHON" ] || PYTHON="`which python`"
+$ [ -n "$PYTHONPATH" ] || PYTHONPATH="$TESTDIR/.." && export PYTHONPATH
+
+TEST FOREACH WITH COMMAND
+
+  $ cp -r $TESTDIR/test_dir_1 test_dir_1
+  $ cd test_dir_1
+  $ git remote add origin file://$TESTDIR/test_dir_1
+  $ cd src/test_dir_2
+  $ git remote add origin file://$TESTDIR/test_dir_1/src/test_dir_2
```

Comment:
Either in this misterious repo these subdirectories are in turn submodules (which I hope not), or you keep adding different remotes as origin of the same repo. These will fail.